lib/ingress/ingress-props.ts

```diff
@@ -0,0 +1,44 @@
+export interface ServiceProps {
+  readonly name: string;
+  readonly port: number;
+  readonly namespace: string;
+  readonly weight: number;
+}
+
+export interface IngressProps {
+  /**
+   * Whether this is an internal service (behind VPN).
+   * @default false
+   */
+  readonly internal?: boolean;
+
+  /**
+   * Custom selector labels, they will be merged with the default app, role, and instance.
+   * They will be applied to the workload, the pod and the service.
+   * @default { app: "<app label from chart>", role: "server", instance: "<construct id>" }
+   */
+  readonly selectorLabels?: { [key: string]: string };
+
+  /**
+   * Domain name for TLS certificate discovery.
+   * @see https://kubernetes-sigs.github.io/aws-load-balancer-controller/v2.3/guide/ingress/cert_discovery/
+   */
+  readonly tlsDomain?: string | string[];
```

Comment:
I wonder if migration to this Ingress is also an opportunity to move away from certificate discovery, and just use cert ARNs. So maybe we should instead have a property for `certificateArn`
```suggestion
  /**
   * ARN of one or more certificate managed by AWS Certificate Manager
   */
  readonly certificateArn?: string | string[];
```

Reply:
Agree with this - makes moving certificates around a lot easier